src/gluonts/nursery/autogluon_tabular.py

```diff
@@ -0,0 +1,118 @@
+from gluonts.model.predictor import Localizer
+from gluonts.model.estimator import Estimator
+
+if TYPE_CHECKING:  # avoid circular import
+    from gluonts.model.estimator import Estimator
+
+OutputTransform = Callable[[DataEntry, np.ndarray], np.ndarray]
+
+
+def get_prediction_dataframe(series):
+    hour_of_day = series.index.hour
+    month_of_year = series.index.month
+    day_of_week = series.index.dayofweek
+    year_idx = series.index.year
+    target = series.values
+    cal = calendar()
+    holidays = cal.holidays(start=series.index.min(), end=series.index.max())
+    df = pd.DataFrame(
+        zip(
+            year_idx,
+            month_of_year,
+            day_of_week,
+            hour_of_day,
+            series.index.isin(holidays),
+            target,
+        ),
+        columns=[
+            "year_idx",
+            "month_of_year",
+            "day_of_week",
+            "hour_of_day",
+            "holiday",
+            "target",
+        ],
+    )
+    convert_type = {x: "category" for x in df.columns.values[:4]}
+    df = df.astype(convert_type)
+    return df
+
+
+class TabularPredictor(Predictor):
+    def __init__(self, ag_model, freq: str, prediction_length: int,) -> None:
+        self.ag_model = ag_model  # task?
+        self.freq = freq
+        self.prediction_length = prediction_length
+
+    def predict(
+        self, dataset: Iterable[Dict], model=None, as_pandas=False
+    ) -> Iterator[SampleForecast]:
+        for idx in range(len(list(dataset))):
+            ts = to_pandas(list(dataset)[idx])
+            df = get_prediction_dataframe(ts)
+            output = self.ag_model.predict(df)
+            yield self.to_forecast(dataset, output, idx)
+
+    def to_forecast(
+        self, gluonts_test: Dataset, y_pred, i: int
+    ) -> Iterator[SampleForecast]:
+        test = [list(gluonts_test)[i]]
+        forecasts = []
+        t = test[i]
+        start = to_pandas(t).index[-self.prediction_length]
+        y_hat = np.array(
+            y_pred[
+                i * self.prediction_length : (i + 1) * self.prediction_length
+            ]
+        )
+        samples = np.vstack([y_hat])
+        sample = SampleForecast(
+            freq=self.freq,
+            start_date=pd.Timestamp(start, freq=self.freq),
+            item_id=str(i),
+            samples=samples,
+        )
+        forecasts.append(sample)
+        return forecasts
+
+
+class TabularEstimator(Estimator):
+    def __init__(self, freq: str, prediction_length: int) -> None:
+        super().__init__()
+        self.task = task
+        self.freq = freq
+        self.prediction_length = prediction_length
+
+    def train(self, training_data: Dataset) -> TabularPredictor:
+        # every time there is only one time series passed
+        # list(training_data)[0] is essentially getting the only time series
+        df = get_prediction_dataframe(
+            to_pandas(list(training_data)[0])[: -self.prediction_length]
+        )
```

Comment:
The whole `training_data` should be used for training here, no slicing.

Also, check my [previous comment](https://github.com/awslabs/gluon-ts/pull/1105#issuecomment-723858720): each time series from the training data should be turned into a dataframe, and these should be stacked into a single one to be passed to `self.task.fit`

Reply:
```
predictor = LocalTabularPredictor(...)
forecasts = predictor.predict(inference_dataset)
```

> equivalent to:
> for ts in inference_dataset:
>      predictor_single = tabular_estimator.train([ts])
>      yield from predictor_single.predict([ts])

There is a localizer in the for loop. So actually only one time series is passed into train each time. Do you mean I should do a "fake" stacking to prevent the case that a Localizer is not used?

Reply:
> Do you mean I should do a "fake" stacking to prevent the case that a Localizer is not used?

Exactly. One can in principle use the estimator with more than one time series to learn a global model, provided that the right features are used: that's not the case now, and so we use the localizer trick, but this way the estimator is multiple-time-series-ready.